code/common/src/main/scala/com/avast/cloud/datadog4s/common/Repeated.scala

```diff
@@ -0,0 +1,34 @@
+package com.avast.cloud.datadog4s.common
+
+import java.time.Duration
+
+import cats.effect.{ Concurrent, Resource, Timer }
+import cats.syntax.applicativeError._
+import cats.syntax.flatMap._
+import cats.syntax.apply._
+import cats.syntax.applicative._
+
+import scala.concurrent.duration._
+
+object Repeated {
+  def run[F[_]: Concurrent: Timer](
+    initDelay: Duration,
```

Comment:
Do we really need initial timeout? I would use the `delay` to keep things simple. Use can always set the value initially manually.

Reply:
true